Zend/zend_ini.c

```diff
@@ -210,7 +211,10 @@ ZEND_API zend_result zend_register_ini_entries(const zend_ini_entry_def *ini_ent
 	 * lead to death.
 	 */
 	if (directives != EG(ini_directives)) {
+		ZEND_ASSERT(module_type == MODULE_TEMPORARY);
```

Comment:
@arnaud-lb just stumbled on this when fixing FPM extension loading using `php_admin_value[extension]` . It actaully use `php_dl` and initializes permanent module so this is actually going to work so I had to remove this assertion in https://github.com/php/php-src/pull/12277 . FPM is single threaded so it should not impact this issue - it just really need to remove the assertion.

There might be potentially other issue with using dl for permanent modules and it has got some limitation but think the current design is really an issue which will require some bigger changes in FPM to get it fixed correctly (pool manager).

Anyway if you see some bigger problem with my change, please comment on that PR.